caesar_cypher.cpp

```diff
@@ -1,9 +1,74 @@
 #include <cstring>
+#include <exception>
 #include <iostream>
+#include <regex>
  
-int main()
+const uint8_t CYPHER_SHIFT_KEY_ARGUMENT_INDEX = 1;
+
+uint8_t calculateEncryptedLetterPosition(uint8_t letter, uint32_t shiftKey)
 {
-	std::string inputString;
-	std::getline(std::cin, inputString);
+    if (' ' == letter)
+    {
+        return (uint8_t)letter;
+    }
+
+    if (('A' <= letter && 'Z' >= letter) ||
+        ('a' <= letter && 'z' >= letter))
+    {
+        const uint8_t alphabetASCIIStarterPosition = ('Z' >= letter) ? ('A' - 1) : ('a' - 1);
+
+        const uint8_t classicLetterAlphabeticPosition = letter - alphabetASCIIStarterPosition;
+        const uint8_t shiftedLetterPosition = (classicLetterAlphabeticPosition + shiftKey) % 26;
+
+        return (shiftedLetterPosition + alphabetASCIIStarterPosition);
+    }
+    else
+    {
+        throw std::invalid_argument("Invalid character provided");
+    }
+}
+
+int main(int argc, char* argv[])
+{
+    try
+    {
+	if (argc < 2)
+        {
+            throw std::invalid_argument("No cypher key provided");
+        }
+        const uint32_t commandLineArgument = (uint32_t)*argv[CYPHER_SHIFT_KEY_ARGUMENT_INDEX];
+
+        if (0 != commandLineArgument)
```

Comment:
In the future try to use boost::program_options. This library provides easy API for cmd argument parsing